### PR TITLE
chore(docs): initial description and examples of watch

### PR DESCRIPTION
**Pull request by RikuVan**

### CODE REVIEW

File: docs/utils/watch.mdx [hunk 1]
```diff
@@ -5,3 +5,52 @@ description: 'watch for changes.'
 ---
 
 # watch
+
+This utility behaves similarly to `useEffect` in React. Proxy objects are tracking with a `get` function passed to the callback. Notice the callback will run eagerly as soon as a proxy is tracked, even if not yet mutated.
```
Comment (dai-shi) [inline docs/utils/watch.mdx:9]: Never thought about the similarity to useEffect. Not sure how others think.

File: docs/utils/watch.mdx [hunk 2]
```diff
@@ -5,3 +5,52 @@ description: 'watch for changes.'
...
+setTimeout(() => {
+  userState.user.name = "Anonymous"
+}, 200)
+// 200ms -> 'Anonymous's session is expired"
+// 'Cleaning up' logged
+```
+
+#### Gotchas
+
+If you remove the setTimeout in the example above, 'Juuso's session is expired' will become 'Anonymous's session is expired', logged twice now. Valtio will batch updates by default. You may pass `{sync: true}` as a second argument to `watch` to disable batching. However, this is relatively untested and should be used with caution.
```
Comment (dai-shi) [inline docs/utils/watch.mdx:55]: "logged twice now instead of three times." could be clearer?

**Review by dai-shi — COMMENTED** [review]
added comments, but i'm fine to merge it as is.

File: docs/utils/watch.mdx (hunk 2 above)
Comment (RikuVan) [reply docs/utils/watch.mdx:55]: Not sure I undersand your comment - it was not logged three times before.

The change is with a timeout you get:
> Juuso's session is expired
> Anonymous's session is expired

But without the timeout you get
> Anonymous's session is expired
> Anonymous's session is expired

**Review by dai-shi — APPROVED** [review]
Merging this for now. Feel free to open new PRs if anyone has ideas to improve it.

**Pull request merged**